File: workspace_cpp/BooleanGraph/src/boolean/nodes/boolean_and.cpp
/*
 * BooleanAND.cpp
 *
 *  Created on: 5 de set de 2015
 *      Author: vroth
 */

#include "boolean_and.hpp"

BooleanAND::BooleanAND(BooleanNode * right, BooleanNode * left) : right(right), left(left) { btype = BOOLEAN_AND; }

BooleanAND::~BooleanAND() { }

std::string BooleanAND::toString() {
	return std::string("(" + left->toString() + " && " + right->toString() + ")");
}

bool BooleanAND::compute(BooleanStore *bs) {
	return left->compute(bs) && right->compute(bs);
}

File: workspace_java/BooleanExpression/src/br/unicamp/ic/bool/expression/BooleanAND.java
package br.unicamp.ic.bool.expression;

import br.unicamp.ic.bool.BooleanStore;

public class BooleanAND extends BooleanNode{
	private BooleanNode left, right;
	
	public BooleanAND(BooleanNode left, BooleanNode right)
	{
		super(bool_type.AND);
		this.left = left;
		this.right = right;
	}
	
	public BooleanNode getLeft()
	{
		return left;
	}
	
	public BooleanNode getRight()
	{
		return right;
	}
	
	public boolean compute(BooleanStore bs)
	{
		return (left.compute(bs) && right.compute(bs));
	}
	
	public String toString()
	{
		return "(" + left.toString() + " && " + right.toString() + ")";
	}
}
